### PR TITLE
Update Java to 1.8 and Gradle to 5.4.1

### PR DESCRIPTION
**Pull request by SpOOnman**
Planuję wydać Sputnika 2.0 i poczyścić też Issues. :) :sunglasses: 
Ta zmiana to trochę zmian w Gradle i głównie update do JUnita 5.

### CODE REVIEW

File: src/main/java/pl/touk/sputnik/connector/stash/json/Anchor.java [hunk 1]
```diff
@@ -16,8 +16,8 @@
 @Builder
 public class Anchor {
     public int line;
-    public String lineType = "CONTEXT";
-    public String fileType = "TO";
+    public String lineType;
```
Comment (sputnikci) [inline src/main/java/pl/touk/sputnik/connector/stash/json/Anchor.java]: \[Checkstyle\] ERROR: Variable 'lineType' must be private and have accessor methods\.

File: src/main/java/pl/touk/sputnik/connector/stash/json/Anchor.java [hunk 2]
```diff
@@ -16,8 +16,8 @@
...
+    public String fileType;
```
Comment (sputnikci) [inline src/main/java/pl/touk/sputnik/connector/stash/json/Anchor.java]: \[Checkstyle\] ERROR: Variable 'fileType' must be private and have accessor methods\.

File: src/test/java/pl/touk/sputnik/CliOptionsTest.java [hunk 3]
```diff
@@ -1,22 +1,23 @@
...
 import pl.touk.sputnik.configuration.CliOption;
 import pl.touk.sputnik.configuration.CliWrapper;
 
-public class CliOptionsTest {
+import static pl.touk.sputnik.SputnikAssertions.assertThat;
 
-    private CliWrapper fixture = new CliWrapper();
+class CliOptionsTest {
+
+    private final static String SAMPLE_CONFIG = "/home/spoonman/sputnik/conf.properties";
```
Comment (sputnikci) [inline src/test/java/pl/touk/sputnik/CliOptionsTest.java]: \[Checkstyle\] ERROR: 'static' modifier out of order with the JLS suggestions\.

File: src/test/java/pl/touk/sputnik/CliOptionsTest.java [hunk 4]
```diff
@@ -1,22 +1,23 @@
...
+    private final static String SAMPLE_CHANGE_ID = "I0a2afb7ae4a94ab1ab473ba00e2ec7de381799a0";
```
Comment (sputnikci) [inline src/test/java/pl/touk/sputnik/CliOptionsTest.java]: \[Checkstyle\] ERROR: 'static' modifier out of order with the JLS suggestions\.

File: src/test/java/pl/touk/sputnik/CliOptionsTest.java [hunk 5]
```diff
@@ -1,22 +1,23 @@
...
+    private final static String SAMPLE_REVISION_ID = "3f37692af2290e8e3fd16d2f43701c24346197f0";
```
Comment (sputnikci) [inline src/test/java/pl/touk/sputnik/CliOptionsTest.java]: \[Checkstyle\] ERROR: 'static' modifier out of order with the JLS suggestions\.

File: src/test/java/pl/touk/sputnik/CliOptionsTest.java [hunk 6]
```diff
@@ -1,22 +1,23 @@
...
+    private final static String SAMPLE_PULL_REQUEST_ID = "123";
```
Comment (sputnikci) [inline src/test/java/pl/touk/sputnik/CliOptionsTest.java]: \[Checkstyle\] ERROR: 'static' modifier out of order with the JLS suggestions\.

File: src/test/java/pl/touk/sputnik/TestEnvironment.java [hunk 7]
```diff
@@ -33,11 +33,11 @@ protected Review review(String filename) {
         return new Review(ImmutableList.of(new ReviewFile(Resources.getResource(filename).getFile())), formatter);
     }
 
-    protected Review nonexistantReview() {
+    protected Review nonExistentReview() {
         return new Review(ImmutableList.of(new ReviewFile("test")), formatter);
     }
 
-    protected Review nonexistantReview(String filename){
+    protected Review nonExistentReview(String filename){
```
Comment (sputnikci) [inline src/test/java/pl/touk/sputnik/TestEnvironment.java]: \[Checkstyle\] ERROR: '\{' is not preceded with whitespace\.

File: src/test/java/pl/touk/sputnik/connector/http/SslVerificationTest.java [hunk 8]
```diff
@@ -57,9 +65,9 @@ public void doNotVerifySslTrustWhenVerificationIsOff() throws Exception {
     }
 
     @Test
-    public void doNotVerifyHostnameWhenVerificationIsOff() throws Exception {
+    void doNotVerifyHostnameWhenVerificationIsOff() throws Exception {
         // given
-        ConnectorDetails connectorDetails = buildConnectorDetails("127.0.0.1", trustedService.httpsPort(), "true", "false");
+        ConnectorDetails connectorDetails = buildConnectorDetails("127.0.0.1", trustedServer.httpsPort(), "true", "false");
```
Comment (sputnikci) [inline src/test/java/pl/touk/sputnik/connector/http/SslVerificationTest.java]: \[PMD\] INFO: Do not hard code the IP address

File: src/test/java/pl/touk/sputnik/connector/http/SslVerificationTest.java [hunk 9]
```diff
@@ -88,49 +96,49 @@ public void verifySslTrust() throws Exception {
...
+        assertThat(thrown).isInstanceOf(SSLHandshakeException.class);
     }
 
     @Test
-    public void verifyHostnameThrowsSSLPeerUnverifiedExceptionWhenHostDoesNotMatch() throws Exception {
+    void verifyHostnameThrowsSSLPeerUnverifiedExceptionWhenHostDoesNotMatch() throws Exception {
         // given
         setSystemTrustStore();
-        ConnectorDetails connectorDetails = buildConnectorDetails("127.0.0.1", trustedService.httpsPort(), "true", "true");
+        ConnectorDetails connectorDetails = buildConnectorDetails("127.0.0.1", trustedServer.httpsPort(), "true", "true");
```
Comment (sputnikci) [inline src/test/java/pl/touk/sputnik/connector/http/SslVerificationTest.java]: \[PMD\] INFO: Do not hard code the IP address

File: build.gradle [hunk 10]
```diff
@@ -133,17 +141,28 @@ dependencies {
     // dependency only for obtaining default configuration
     compile 'io.gitlab.arturbosch.detekt:detekt-cli:1.0.0.RC3'
 
+    compileOnly 'org.projectlombok:lombok:1.18.4'
```
Comment (alien11689) [inline build.gradle]: do yo really have to add lombok as 4 dependencies?

File: build.gradle [hunk 11]
```diff
@@ -171,7 +190,7 @@ task copyToLib(type: Copy) {
 
 artifacts {
     archives jar
-    archives shadowJar
+//    archives shadowJar
```
Comment (alien11689) [inline build.gradle]: remove if not needed

File: src/main/java/pl/touk/sputnik/connector/stash/StashFacade.java [hunk 12]
```diff
@@ -2,23 +2,16 @@
...
 import pl.touk.sputnik.connector.ConnectorFacade;
 import pl.touk.sputnik.connector.Connectors;
-import pl.touk.sputnik.connector.stash.json.Anchor;
-import pl.touk.sputnik.connector.stash.json.Comment;
-import pl.touk.sputnik.connector.stash.json.DiffSegment;
-import pl.touk.sputnik.connector.stash.json.FileComment;
-import pl.touk.sputnik.connector.stash.json.LineComment;
-import pl.touk.sputnik.connector.stash.json.LineSegment;
-import pl.touk.sputnik.connector.stash.json.ReviewElement;
+import pl.touk.sputnik.connector.stash.json.*;
```
Comment (alien11689) [inline src/main/java/pl/touk/sputnik/connector/stash/StashFacade.java]: imports without star were better

File: src/main/java/pl/touk/sputnik/connector/stash/StashFacade.java [hunk 13]
```diff
@@ -133,12 +126,10 @@ private FileComment toFileComment(String key, pl.touk.sputnik.review.Comment com
...
 
-    private <T> List<T> transform(List<JSONObject> jsonList, Class<T> someClass) {
-        List<T> result = Lists.newArrayList();
+    private <T> List<T> transform(JSONArray jsonList, Class<T> someClass) {
+        List<T> result;
         try {
-            for (JSONObject jsonObject : jsonList) {
-                result.add(objectMapper.readValue(jsonObject.toJSONString(), someClass));
-            }
+            result = objectMapper.readValue(jsonList.toJSONString(), objectMapper.getTypeFactory().constructCollectionType(List.class, someClass));
```
Comment (alien11689) [inline src/main/java/pl/touk/sputnik/connector/stash/StashFacade.java]: if you extract try/catch as method then you won't need result variable declaration without assigment

File: src/main/java/pl/touk/sputnik/processor/spotbugs/SpotBugsProcessor.java [hunk 14]
```diff
@@ -1,18 +1,11 @@
 package pl.touk.sputnik.processor.spotbugs;
 
-import edu.umd.cs.findbugs.ClassScreener;
-import edu.umd.cs.findbugs.DetectorFactoryCollection;
-import edu.umd.cs.findbugs.FindBugs2;
-import edu.umd.cs.findbugs.IClassScreener;
-import edu.umd.cs.findbugs.Priorities;
-import edu.umd.cs.findbugs.Project;
+import edu.umd.cs.findbugs.*;
```
Comment (alien11689) [inline src/main/java/pl/touk/sputnik/processor/spotbugs/SpotBugsProcessor.java]: imports without star were better

File: src/test/java/pl/touk/sputnik/CliOptionsTest.java (hunk 3 above)
Comment (alien11689) [reply src/test/java/pl/touk/sputnik/CliOptionsTest.java]: this suggestion is worth applying

File: src/test/java/pl/touk/sputnik/processor/codenarc/CodeNarcProcessorTest.java [hunk 15]
```diff
@@ -1,22 +1,17 @@
...
+import org.junit.jupiter.api.Test;
 import pl.touk.sputnik.configuration.Configuration;
 import pl.touk.sputnik.configuration.ConfigurationBuilder;
-import pl.touk.sputnik.review.Review;
-import pl.touk.sputnik.review.ReviewFile;
-import pl.touk.sputnik.review.ReviewFormatterFactory;
-import pl.touk.sputnik.review.ReviewResult;
-import pl.touk.sputnik.review.Severity;
-import pl.touk.sputnik.review.Violation;
+import pl.touk.sputnik.review.*;
```
Comment (alien11689) [inline src/test/java/pl/touk/sputnik/processor/codenarc/CodeNarcProcessorTest.java]: explicit imports were better

File: src/test/java/pl/touk/sputnik/processor/codenarc/CodeNarcReviewProcessorFactoryTest.java [hunk 16]
```diff
@@ -1,34 +1,33 @@
...
 
     @Test
-    public void testCreate() throws Exception {
+    void testCreate() {
         Configuration configuration = mock(Configuration.class);
         when(configuration.getProperty(eq(GeneralOption.CODE_NARC_ENABLED))).thenReturn("true");
 
         ReviewProcessorFactory factory = new CodeNarcReviewProcessorFactory();
-        assertNotNull(factory.create(configuration));
+        assertThat(factory.create(configuration)).isNotNull();
```
Comment (alien11689) [inline src/test/java/pl/touk/sputnik/processor/codenarc/CodeNarcReviewProcessorFactoryTest.java]: is it really readable?

File: src/test/java/pl/touk/sputnik/processor/detekt/DetektProcessorTest.java [hunk 17]
```diff
@@ -1,22 +1,17 @@
...
+import org.junit.jupiter.api.Test;
 import pl.touk.sputnik.configuration.Configuration;
 import pl.touk.sputnik.configuration.ConfigurationBuilder;
-import pl.touk.sputnik.review.Review;
-import pl.touk.sputnik.review.ReviewFile;
-import pl.touk.sputnik.review.ReviewFormatterFactory;
-import pl.touk.sputnik.review.ReviewResult;
-import pl.touk.sputnik.review.Severity;
-import pl.touk.sputnik.review.Violation;
+import pl.touk.sputnik.review.*;
```
Comment (alien11689) [inline src/test/java/pl/touk/sputnik/processor/detekt/DetektProcessorTest.java]: i won't write more comments for star imports... :)

File: src/test/java/pl/touk/sputnik/processor/scalastyle/ScalastyleReviewProcessorFactoryTest.java [hunk 18]
```diff
@@ -1,33 +1,32 @@
...
 
     @Test
-    public void testIsEnabled() throws Exception {
+    void testIsEnabled() {
         Configuration configuration = mock(Configuration.class);
         when(configuration.getProperty(eq(GeneralOption.SCALASTYLE_ENABLED))).thenReturn("true");
 
         ReviewProcessorFactory factory = new ScalastyleReviewProcessorFactory();
-        assertTrue(factory.isEnabled(configuration));
+        assertThat(factory.isEnabled(configuration)).isTrue();
```
Comment (alien11689) [inline src/test/java/pl/touk/sputnik/processor/scalastyle/ScalastyleReviewProcessorFactoryTest.java]: is it more readable?

File: src/test/java/pl/touk/sputnik/processor/sonar/SonarProcessorTest.java [hunk 19]
```diff
@@ -16,10 +16,11 @@
 
 import static org.assertj.core.api.Assertions.assertThat;
 
-public class SonarProcessorTest extends TestEnvironment {
+class SonarProcessorTest extends TestEnvironment {
 
     @Test
-    public void shouldFilterResultFiles() {
+    void shouldFilterResultFiles() {
+        //given
```
Comment (alien11689) [inline src/test/java/pl/touk/sputnik/processor/sonar/SonarProcessorTest.java]: why do you add this comments? You didn't do this to many tests above. IMO It's better to not have those comments at all

**Comment by alien11689** [comment]
This PR is too big for me...

File: src/main/java/pl/touk/sputnik/connector/stash/json/Anchor.java (hunk 1 above)
Comment (SpOOnman) [reply src/main/java/pl/touk/sputnik/connector/stash/json/Anchor.java]: Done

File: src/test/java/pl/touk/sputnik/CliOptionsTest.java (hunk 3 above)
Comment (SpOOnman) [reply src/test/java/pl/touk/sputnik/CliOptionsTest.java]: Done

File: src/test/java/pl/touk/sputnik/TestEnvironment.java (hunk 7 above)
Comment (SpOOnman) [reply src/test/java/pl/touk/sputnik/TestEnvironment.java]: Done

File: src/test/java/pl/touk/sputnik/connector/http/SslVerificationTest.java (hunk 9 above)
Comment (SpOOnman) [reply src/test/java/pl/touk/sputnik/connector/http/SslVerificationTest.java]: Done

File: build.gradle [hunk 20]
```diff
@@ -20,18 +20,24 @@ apply plugin: 'coveralls'
 apply plugin: 'maven'
 apply plugin: 'signing'
 apply plugin: 'pl.allegro.tech.build.axion-release'
-apply plugin: 'com.github.johnrengelman.shadow'
+//apply plugin: 'com.github.johnrengelman.shadow'
```
[the remark this answers is not in the document]
Comment (SpOOnman) [reply build.gradle]: I need to fix shadow gradle task in next commits before release.

File: build.gradle (hunk 10 above)
Comment (SpOOnman) [reply build.gradle]: This is the way Gradle 5 takes case or annotation processing.

File: src/main/java/pl/touk/sputnik/connector/stash/StashFacade.java (hunk 12 above)
Comment (SpOOnman) [reply src/main/java/pl/touk/sputnik/connector/stash/StashFacade.java]: I've optimised all imports and remove all stars! :star2:

File: src/main/java/pl/touk/sputnik/connector/stash/StashFacade.java (hunk 13 above)
Comment (SpOOnman) [reply src/main/java/pl/touk/sputnik/connector/stash/StashFacade.java]: Done

File: src/main/java/pl/touk/sputnik/processor/spotbugs/SpotBugsProcessor.java (hunk 14 above)
Comment (SpOOnman) [reply src/main/java/pl/touk/sputnik/processor/spotbugs/SpotBugsProcessor.java]: Done

File: src/test/java/pl/touk/sputnik/processor/codenarc/CodeNarcProcessorTest.java (hunk 15 above)
Comment (SpOOnman) [reply src/test/java/pl/touk/sputnik/processor/codenarc/CodeNarcProcessorTest.java]: Done

File: src/test/java/pl/touk/sputnik/processor/detekt/DetektProcessorTest.java (hunk 17 above)
Comment (SpOOnman) [reply src/test/java/pl/touk/sputnik/processor/detekt/DetektProcessorTest.java]: Done

File: src/test/java/pl/touk/sputnik/processor/codenarc/CodeNarcReviewProcessorFactoryTest.java (hunk 16 above)
Comment (SpOOnman) [reply src/test/java/pl/touk/sputnik/processor/codenarc/CodeNarcReviewProcessorFactoryTest.java]: I think it is. I prefer assertj over junit assertions.

File: src/test/java/pl/touk/sputnik/processor/scalastyle/ScalastyleReviewProcessorFactoryTest.java (hunk 18 above)
Comment (SpOOnman) [reply src/test/java/pl/touk/sputnik/processor/scalastyle/ScalastyleReviewProcessorFactoryTest.java]: I think it is.

File: src/test/java/pl/touk/sputnik/processor/sonar/SonarProcessorTest.java (hunk 19 above)
Comment (SpOOnman) [reply src/test/java/pl/touk/sputnik/processor/sonar/SonarProcessorTest.java]: Historically there were added by developers. With release 2.0 I want to remove them too. I will do it in next commits.

**Pull request merged**